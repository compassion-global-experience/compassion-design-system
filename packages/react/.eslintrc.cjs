module.exports = {
  extends: [
    'standard-jsx',
    'standard-react',
    'plugin:jsx-a11y/recommended',
    'plugin:react/recommended',
    'plugin:react/jsx-runtime',
    'plugin:react-hooks/recommended',
    'plugin:storybook/recommended',
  ],
  env: {
    browser: true,
    es6: true,
    jest: true,
    node: false,
  },
  rules: {
    'jsx-quotes': ['error', 'prefer-double'],
  },
};
