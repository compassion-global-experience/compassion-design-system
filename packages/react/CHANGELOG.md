# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# 0.1.0 (2023-04-08)

### Features

- **react:** css usage and documentation ([11e317c](https://github.com/compassion-global-experience/compassion-design-system/commit/11e317cca4ecdaa385e86d3a5571d7dca554685d))
- **react:** re-export css package ([67549a5](https://github.com/compassion-global-experience/compassion-design-system/commit/67549a58de3de6e02b58c5bdb6f4605001aae783))
- **react:** switch between themes in storybook ([79119ba](https://github.com/compassion-global-experience/compassion-design-system/commit/79119ba48ed3410eaebe9a91ccd076b05c2580c1))
