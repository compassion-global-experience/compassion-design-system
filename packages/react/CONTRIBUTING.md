# Contributing to the `react-web` Package

The `react` package is structured with a component-per-directory convention
inside the `src` directory. Each component directory generally contains 3 items:

- A **TSX file** that describes the component‚Äôs structure and behavior
- A **.stories.ts file** that illustrates most of the ways the component can
  be configured
- An **index.ts** file that exports the TSX component‚Äôs contents

## Workflow

The package uses [Storybook](https://storybook.js.org/docs/react/get-started/introduction)
to facilitate component development and documentation. Storybook uses the files
mentioned above to render components in their various states, and serves as
documentation of those components by surfacing code comments, `propTypes`, and
`defaultProps` values.

To launch Storybook, make sure you‚Äôre in the [/packages/react](/packages/react)
directory and run the command `npm run storybook`. After the project is built,
Storybook will launch in your default browser. New components or modifications
will be displayed in real time! üéâ

The configurations outlined in each component‚Äôs .stories.ts file can be seen in
Storybook‚Äôs **Docs** tab.

## Commit Messages & Versioning

Every package in the compassion-gds monorepo adheres to
[semantic versioning](https://semver.org/). In order to simplify the versioning
and release process, please be sure your commit messages adhere to the
[‚ÄúConventional Commits‚Äù pattern](https://www.conventionalcommits.org/en/v1.0.0/#summary).
You may wish to install the [commitizen/cz-cli](https://github.com/commitizen/cz-cli)
to make this near-bulletproof.

## Making a Pull Request

Once your changes are ready, open a pull request against the `development`
branch of the project, and it will be merged in as quickly as possible!
