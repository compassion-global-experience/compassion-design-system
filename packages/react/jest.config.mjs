/*
 * For a detailed explanation regarding each configuration property, visit:
 * https://jestjs.io/docs/configuration
 */

export default {
  // A preset that is used as a base for Jest's configuration
  preset: 'ts-jest/presets/default-esm',

  // Automatically clear mock calls, instances, contexts and results before every test
  clearMocks: true,

  // Indicates whether the coverage information should be collected while executing the test
  collectCoverage: true,

  // The directory where Jest should output its coverage files
  coverageDirectory: "coverage",

  // Indicates which provider should be used to instrument code for coverage
  coverageProvider: "v8",

  // The test environment that will be used for testing
  testEnvironment: "jsdom",

  // A map from regular expressions to module names or to arrays of module names that allow to stub out resources with a single module
  moduleNameMapper: {
    ".(css|less|scss)$": "identity-obj-proxy",
  },

  // Whether to use watchman for file crawling
  watchman: true,
};
