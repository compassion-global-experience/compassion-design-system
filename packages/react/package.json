{
  "name": "@compassion-gds/react",
  "version": "0.2.0-alpha.2",
  "description": "",
  "type": "module",
  "main": "dist/cjs/index.js",
  "module": "dist/esm/index.js",
  "files": [
    "dist"
  ],
  "types": "dist/index.d.ts",
  "exports": {
    ".": {
      "browser": {
        "default": "./dist/esm/index.js"
      },
      "import": "./dist/esm/index.js",
      "require": "./dist/cjs/index.js"
    },
    "./css/base.css": "./dist/css/reset.css",
    "./css/light.css": "./dist/css/cds-light.css",
    "./css/dark.css": "./dist/css/cds-dark.css",
    "./css/*": "./dist/css/*"
  },
  "scripts": {
    "dev": "rollup -c -w",
    "copy-css": "bash ./scripts/copy-css.sh",
    "prepare": "rollup -c && npm run copy-css",
    "test": "jest",
    "storybook": "storybook dev -p 6007",
    "build:storybook": "storybook build --quiet",
    "percy": "cross-env PERCY_TOKEN=$PERCY_TOKEN_REACT percy storybook ./storybook-static",
    "percy-snapshots": "npm run build:storybook && npm run percy"
  },
  "repository": {
    "type": "git",
    "url": "https://github.com/compassion-global-experience/compassion-design-system.git",
    "directory": "packages/react"
  },
  "keywords": [
    "compassion",
    "compassion-gds",
    "gds",
    "react"
  ],
  "author": "Compassion XD",
  "license": "MIT",
  "bugs": {
    "url": "https://github.com/compassion-global-experience/compassion-design-system/issues"
  },
  "homepage": "https://github.com/compassion-global-experience/compassion-design-system#readme",
  "devDependencies": {
    "@babel/core": "^7.20.12",
    "@babel/preset-env": "^7.20.2",
    "@babel/preset-react": "^7.18.6",
    "@babel/preset-typescript": "^7.18.6",
    "@compassion-gds/core": "^0.1.0-alpha.0",
    "@rollup/plugin-commonjs": "^24.0.1",
    "@rollup/plugin-node-resolve": "^15.0.1",
    "@rollup/plugin-replace": "^5.0.2",
    "@rollup/plugin-typescript": "^11.1.0",
    "@storybook/addon-actions": "^7.0.2",
    "@storybook/addon-docs": "^7.0.2",
    "@storybook/addon-essentials": "^7.0.2",
    "@storybook/addon-interactions": "^7.0.2",
    "@storybook/addon-links": "^7.0.2",
    "@storybook/preset-react-webpack": "7.0.2",
    "@storybook/react": "7.0.2",
    "@storybook/react-webpack5": "7.0.2",
    "@storybook/testing-library": "^0.0.14-next.1",
    "@testing-library/react": "^13.4.0",
    "@types/jest": "^29.4.0",
    "@types/react": "^18.0.27",
    "babel-loader": "^8.3.0",
    "eslint": "^8.33.0",
    "eslint-config-standard-jsx": "^11.0.0",
    "eslint-config-standard-react": "^13.0.0",
    "eslint-plugin-jsx-a11y": "^6.7.1",
    "eslint-plugin-react": "^7.32.2",
    "eslint-plugin-react-hooks": "^4.6.0",
    "eslint-plugin-storybook": "^0.6.10",
    "jest": "^29.4.1",
    "jest-environment-jsdom": "^29.4.1",
    "phosphor-react": "^1.4.1",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "rollup": "^3.20.2",
    "rollup-plugin-dts": "^5.1.1",
    "rollup-plugin-peer-deps-external": "^2.2.4",
    "rollup-plugin-postcss": "^4.0.2",
    "storybook": "7.0.2",
    "ts-jest": "^29.0.5",
    "webpack": "^5.75.0"
  },
  "peerDependencies": {
    "react": "16.8.x || 17.x || 18.x",
    "react-dom": "16.8.x || 17.x || 18.x"
  }
}
