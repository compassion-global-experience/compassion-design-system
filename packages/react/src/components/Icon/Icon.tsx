'use client';

import styles from '@compassion-design-system/core/src/components/Icon/icon.module.css';
import { getClasses } from '../../utils/classes';
import * as icons from '../icons';
import SkippyIcon from '../icons/Skippy';

export const cdsIconMap = {
  arrowClockwise: icons.ArrowClockwise,
  arrowDown: icons.ArrowDown,
  arrowLeft: icons.ArrowLeft,
  arrowRight: icons.ArrowRight,
  arrowUp: icons.ArrowUp,
  arrowUpRight: icons.ArrowUpRight,
  arrowsLeftRight: icons.ArrowsLeftRight,
  browser: icons.Browser,
  camera: icons.Camera,
  caretDown: icons.CaretDown,
  caretLeft: icons.CaretLeft,
  caretRight: icons.CaretRight,
  caretUp: icons.CaretUp,
  chatDots: icons.ChatDots,
  chatText: icons.ChatCenteredText,
  check: icons.Check,
  clockCounterClockwise: icons.ClockCounterClockwise,
  cloudArrowUp: icons.CloudArrowUp,
  cloudArrowDown: icons.CloudArrowDown,
  church: icons.Church,
  compass: icons.Compass,
  envelope: icons.EnvelopeSimple,
  eye: icons.Eye,
  eyeSlash: icons.EyeSlash,
  fileText: icons.FileText,
  gear: icons.Gear,
  gift: icons.Gift,
  globe: icons.GlobeStand,
  info: icons.Info,
  images: icons.Images,
  house: icons.HouseLine,
  link: icons.Link,
  list: icons.List,
  lockKey: icons.LockKey,
  lockSimple: icons.LockSimple,
  magnigyGlass: icons.MagnifyingGlass,
  magnigyGlassPlus: icons.MagnifyingGlassPlus,
  mapPin: icons.MapPin,
  mapTrifold: icons.MapTrifold,
  minus: icons.Minus,
  playCircle: icons.PlayCircle,
  plus: icons.Plus,
  plusCircle: icons.PlusCircle,
  skippy: SkippyIcon,
  spinner: icons.Spinner,
  spinnerGap: icons.SpinnerGap,
  trash: icons.Trash,
  thumbsUp: icons.ThumbsUp,
  user: icons.User,
  warningCircle: icons.WarningCircle,
  warningOctagon: icons.WarningOctagon,
  wheelchair: icons.Wheelchair,
  x: icons.X,
  xCircle: icons.XCircle,
};

export const iconSizeMap = {
  xs: 12,
  sm: 16,
  md: 24,
  lg: 32,
};

export interface IconProps {
  description: string;
  color?: 'default' | 'primary' | 'muted';
  size?: keyof typeof iconSizeMap;
  icon: keyof typeof cdsIconMap;
}

const Icon = (props: IconProps) => {
  const { description, color = 'default', size = 'md', icon } = props;
  const SelectedIcon = cdsIconMap[icon];

  const iconColorClassPrefix = 'cds-icon__svg--';
  const colorClassName = getClasses(styles, [
    `${iconColorClassPrefix}${color}`,
  ]);

  return (
    <>
      {SelectedIcon && (
        <SelectedIcon
          className={colorClassName}
          size={iconSizeMap[size]}
          alt={description}
        />
      )}
    </>
  );
};

export default Icon;
