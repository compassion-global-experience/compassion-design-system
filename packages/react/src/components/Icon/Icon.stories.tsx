import { Meta } from '@storybook/react';
import Icon, { cdsIconMap } from './Icon';

const meta: Meta<typeof Icon> = {
  title: 'Components/Icon',
  component: Icon,
};

export default meta;

const iconDescriptionArgType = { control: 'text' };
const iconArgTypes = {
  control: { type: 'select' },
  options: Object.keys(cdsIconMap).sort(),
};
const sizeArgType = {
  control: { type: 'select' },
  options: ['xs', 'sm', 'md', 'lg'],
};
const colorArgType = {
  control: { type: 'select' },
  options: ['default', 'primary', 'muted', 'white'],
};

export const Default = {
  args: {
    description: 'check mark',
    icon: 'check',
    color: 'default',
    size: 'md',
  },
  argTypes: {
    icon: iconArgTypes,
    description: iconDescriptionArgType,
    size: sizeArgType,
    color: colorArgType,
  },
};
