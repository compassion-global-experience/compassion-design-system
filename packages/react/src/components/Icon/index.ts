import Icon from './Icon';

export {
  ArrowLeft,
  ArrowRight,
  X,
  Check,
  Info,
  CaretDown,
} from 'phosphor-react';

export default Icon;
