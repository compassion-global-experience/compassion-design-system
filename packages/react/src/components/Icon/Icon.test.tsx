import { render } from '@testing-library/react';
import Icon from './Icon';

describe('Icon', () => {
  it('should render with default props', () => {
    const { container } = render(<Icon icon="skippy" description="Skippy" />);
    expect(container).toMatchSnapshot();
  });

  it('should render a large icon', () => {
    const { container } = render(
      <Icon icon="skippy" description="Skippy" size="lg" />,
    );
    expect(container).toMatchSnapshot();
  });

  it('should render a muted icon', () => {
    const { container } = render(
      <Icon icon="skippy" description="Skippy" color="muted" />,
    );
    expect(container).toMatchSnapshot();
  });
});
