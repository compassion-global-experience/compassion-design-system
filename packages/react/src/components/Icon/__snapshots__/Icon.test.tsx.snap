// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should render a large icon 1`] = `
<div>
  <div
    class="cds-icon__svg--default"
    style="width: 32px;"
  >
    <svg
      fill="none"
      role="img"
      viewBox="0 0 16 16"
      width="100%"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        Skippy
      </title>
      <path
        d="M4.37819 2.25418C4.25773 2.41851 4.17169 2.60014 4.12867 2.80079C4.06844 2.99972 4.06844 3.20729 4.12006 3.42525C4.16308 3.63282 4.25773 3.82309 4.37819 3.9978C4.50725 4.17078 4.66212 4.30051 4.86174 4.4216C5.05103 4.53403 5.24892 4.60323 5.47435 4.65512C5.70667 4.69836 5.9321 4.69836 6.17301 4.66377C6.39672 4.61187 6.61354 4.54268 6.81144 4.4216C6.99213 4.30916 7.1556 4.17078 7.28639 3.9978C7.41545 3.83347 7.49289 3.6432 7.53591 3.45119C7.58753 3.25227 7.59614 3.04469 7.54451 2.83539C7.50149 2.61917 7.41545 2.42889 7.28639 2.25418C7.15732 2.08986 6.99385 1.95147 6.80283 1.83039C6.61354 1.71795 6.41565 1.64011 6.19022 1.58822C5.95791 1.54497 5.72387 1.54497 5.49156 1.58822C5.25925 1.62281 5.05103 1.7093 4.85313 1.82174C4.66384 1.93244 4.50725 2.08121 4.37819 2.25418ZM4.28354 7.99013L4.03402 8.36204C4.22331 8.19771 4.5778 7.98149 5.07684 7.71164C5.4124 7.53001 6.23152 7.09584 7.53419 6.42987C8.51678 5.91786 9.32729 5.15675 9.95711 4.14137C10.1636 3.81272 10.3443 3.47368 10.4923 3.11907C10.5783 2.9115 10.6472 2.72987 10.6902 2.58111C10.7074 2.52057 10.716 2.40813 10.716 2.26975C10.7246 2.11407 10.716 1.95839 10.6988 1.83731C10.6816 1.68163 10.6558 1.57784 10.6127 1.52595C10.5611 1.47405 10.5009 1.5 10.4407 1.60379C9.98293 2.23689 9.44087 2.83366 8.80244 3.38027C8.12099 3.95283 7.37071 4.44582 6.54299 4.87134C5.50877 5.38336 4.66384 5.91959 3.9996 6.47485C3.34397 7.01281 2.73996 7.67877 2.18069 8.46755C2.11185 8.57134 2.05162 8.70972 2 8.9C2.08604 8.83946 2.18929 8.82216 2.31835 8.83946L2.12046 9.23731L2.43881 8.97784C2.54206 8.89135 2.63671 8.83081 2.73135 8.80486L2.65391 9.08163C2.86041 8.87405 3.13747 8.67513 3.45582 8.46582L3.34397 8.74259C3.49024 8.59556 3.63651 8.46582 3.78278 8.36204L4.28354 7.99013ZM6.69786 7.14081C6.3537 7.33974 6.04223 7.55596 5.7669 7.79986C5.25753 8.24268 4.83593 8.80486 4.50037 9.4812C4.1648 10.097 3.90496 10.8668 3.69846 11.7957C3.60381 12.2385 3.54358 12.6536 3.52638 13.0515L3.75008 12.7401C3.68125 13.1553 3.63823 13.6154 3.62963 14.1101C3.73287 13.9285 3.80171 13.8074 3.85333 13.7382L3.82752 14.9335L4.31968 13.7382C4.35409 13.842 4.37991 13.9544 4.41432 14.0841C4.71547 13.3819 4.97531 12.8024 5.20763 12.3423C5.43994 11.8493 5.68257 11.4151 5.91489 11.0432C6.13859 10.6713 6.39844 10.3063 6.67377 9.96035C6.92329 9.64899 7.24337 9.27536 7.63055 8.86886C7.69939 8.77372 8.1055 8.5921 8.13131 8.58345C8.3034 8.52291 8.5185 8.45372 8.79555 8.38452C8.92462 8.34993 9.23609 8.28074 9.70931 8.15965C10.1223 8.06452 10.4596 7.96938 10.7177 7.90883C11.5196 7.68396 12.1322 7.44006 12.5711 7.1806C13.132 6.83464 13.459 6.43506 13.5364 5.96802C12.9927 6.08911 12.4506 6.16695 11.8896 6.20154C11.5885 6.21884 11.1393 6.22749 10.5267 6.22749C9.99153 6.22749 9.57853 6.23614 9.28599 6.27073C8.83685 6.32263 8.39804 6.40912 7.97472 6.5648C7.49977 6.72567 7.07817 6.91594 6.69786 7.14081Z"
        fill="currentColor"
      />
    </svg>
  </div>
</div>
`;

exports[`Icon should render a muted icon 1`] = `
<div>
  <div
    class="cds-icon__svg--muted"
    style="width: 24px;"
  >
    <svg
      fill="none"
      role="img"
      viewBox="0 0 16 16"
      width="100%"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        Skippy
      </title>
      <path
        d="M4.37819 2.25418C4.25773 2.41851 4.17169 2.60014 4.12867 2.80079C4.06844 2.99972 4.06844 3.20729 4.12006 3.42525C4.16308 3.63282 4.25773 3.82309 4.37819 3.9978C4.50725 4.17078 4.66212 4.30051 4.86174 4.4216C5.05103 4.53403 5.24892 4.60323 5.47435 4.65512C5.70667 4.69836 5.9321 4.69836 6.17301 4.66377C6.39672 4.61187 6.61354 4.54268 6.81144 4.4216C6.99213 4.30916 7.1556 4.17078 7.28639 3.9978C7.41545 3.83347 7.49289 3.6432 7.53591 3.45119C7.58753 3.25227 7.59614 3.04469 7.54451 2.83539C7.50149 2.61917 7.41545 2.42889 7.28639 2.25418C7.15732 2.08986 6.99385 1.95147 6.80283 1.83039C6.61354 1.71795 6.41565 1.64011 6.19022 1.58822C5.95791 1.54497 5.72387 1.54497 5.49156 1.58822C5.25925 1.62281 5.05103 1.7093 4.85313 1.82174C4.66384 1.93244 4.50725 2.08121 4.37819 2.25418ZM4.28354 7.99013L4.03402 8.36204C4.22331 8.19771 4.5778 7.98149 5.07684 7.71164C5.4124 7.53001 6.23152 7.09584 7.53419 6.42987C8.51678 5.91786 9.32729 5.15675 9.95711 4.14137C10.1636 3.81272 10.3443 3.47368 10.4923 3.11907C10.5783 2.9115 10.6472 2.72987 10.6902 2.58111C10.7074 2.52057 10.716 2.40813 10.716 2.26975C10.7246 2.11407 10.716 1.95839 10.6988 1.83731C10.6816 1.68163 10.6558 1.57784 10.6127 1.52595C10.5611 1.47405 10.5009 1.5 10.4407 1.60379C9.98293 2.23689 9.44087 2.83366 8.80244 3.38027C8.12099 3.95283 7.37071 4.44582 6.54299 4.87134C5.50877 5.38336 4.66384 5.91959 3.9996 6.47485C3.34397 7.01281 2.73996 7.67877 2.18069 8.46755C2.11185 8.57134 2.05162 8.70972 2 8.9C2.08604 8.83946 2.18929 8.82216 2.31835 8.83946L2.12046 9.23731L2.43881 8.97784C2.54206 8.89135 2.63671 8.83081 2.73135 8.80486L2.65391 9.08163C2.86041 8.87405 3.13747 8.67513 3.45582 8.46582L3.34397 8.74259C3.49024 8.59556 3.63651 8.46582 3.78278 8.36204L4.28354 7.99013ZM6.69786 7.14081C6.3537 7.33974 6.04223 7.55596 5.7669 7.79986C5.25753 8.24268 4.83593 8.80486 4.50037 9.4812C4.1648 10.097 3.90496 10.8668 3.69846 11.7957C3.60381 12.2385 3.54358 12.6536 3.52638 13.0515L3.75008 12.7401C3.68125 13.1553 3.63823 13.6154 3.62963 14.1101C3.73287 13.9285 3.80171 13.8074 3.85333 13.7382L3.82752 14.9335L4.31968 13.7382C4.35409 13.842 4.37991 13.9544 4.41432 14.0841C4.71547 13.3819 4.97531 12.8024 5.20763 12.3423C5.43994 11.8493 5.68257 11.4151 5.91489 11.0432C6.13859 10.6713 6.39844 10.3063 6.67377 9.96035C6.92329 9.64899 7.24337 9.27536 7.63055 8.86886C7.69939 8.77372 8.1055 8.5921 8.13131 8.58345C8.3034 8.52291 8.5185 8.45372 8.79555 8.38452C8.92462 8.34993 9.23609 8.28074 9.70931 8.15965C10.1223 8.06452 10.4596 7.96938 10.7177 7.90883C11.5196 7.68396 12.1322 7.44006 12.5711 7.1806C13.132 6.83464 13.459 6.43506 13.5364 5.96802C12.9927 6.08911 12.4506 6.16695 11.8896 6.20154C11.5885 6.21884 11.1393 6.22749 10.5267 6.22749C9.99153 6.22749 9.57853 6.23614 9.28599 6.27073C8.83685 6.32263 8.39804 6.40912 7.97472 6.5648C7.49977 6.72567 7.07817 6.91594 6.69786 7.14081Z"
        fill="currentColor"
      />
    </svg>
  </div>
</div>
`;

exports[`Icon should render with default props 1`] = `
<div>
  <div
    class="cds-icon__svg--default"
    style="width: 24px;"
  >
    <svg
      fill="none"
      role="img"
      viewBox="0 0 16 16"
      width="100%"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        Skippy
      </title>
      <path
        d="M4.37819 2.25418C4.25773 2.41851 4.17169 2.60014 4.12867 2.80079C4.06844 2.99972 4.06844 3.20729 4.12006 3.42525C4.16308 3.63282 4.25773 3.82309 4.37819 3.9978C4.50725 4.17078 4.66212 4.30051 4.86174 4.4216C5.05103 4.53403 5.24892 4.60323 5.47435 4.65512C5.70667 4.69836 5.9321 4.69836 6.17301 4.66377C6.39672 4.61187 6.61354 4.54268 6.81144 4.4216C6.99213 4.30916 7.1556 4.17078 7.28639 3.9978C7.41545 3.83347 7.49289 3.6432 7.53591 3.45119C7.58753 3.25227 7.59614 3.04469 7.54451 2.83539C7.50149 2.61917 7.41545 2.42889 7.28639 2.25418C7.15732 2.08986 6.99385 1.95147 6.80283 1.83039C6.61354 1.71795 6.41565 1.64011 6.19022 1.58822C5.95791 1.54497 5.72387 1.54497 5.49156 1.58822C5.25925 1.62281 5.05103 1.7093 4.85313 1.82174C4.66384 1.93244 4.50725 2.08121 4.37819 2.25418ZM4.28354 7.99013L4.03402 8.36204C4.22331 8.19771 4.5778 7.98149 5.07684 7.71164C5.4124 7.53001 6.23152 7.09584 7.53419 6.42987C8.51678 5.91786 9.32729 5.15675 9.95711 4.14137C10.1636 3.81272 10.3443 3.47368 10.4923 3.11907C10.5783 2.9115 10.6472 2.72987 10.6902 2.58111C10.7074 2.52057 10.716 2.40813 10.716 2.26975C10.7246 2.11407 10.716 1.95839 10.6988 1.83731C10.6816 1.68163 10.6558 1.57784 10.6127 1.52595C10.5611 1.47405 10.5009 1.5 10.4407 1.60379C9.98293 2.23689 9.44087 2.83366 8.80244 3.38027C8.12099 3.95283 7.37071 4.44582 6.54299 4.87134C5.50877 5.38336 4.66384 5.91959 3.9996 6.47485C3.34397 7.01281 2.73996 7.67877 2.18069 8.46755C2.11185 8.57134 2.05162 8.70972 2 8.9C2.08604 8.83946 2.18929 8.82216 2.31835 8.83946L2.12046 9.23731L2.43881 8.97784C2.54206 8.89135 2.63671 8.83081 2.73135 8.80486L2.65391 9.08163C2.86041 8.87405 3.13747 8.67513 3.45582 8.46582L3.34397 8.74259C3.49024 8.59556 3.63651 8.46582 3.78278 8.36204L4.28354 7.99013ZM6.69786 7.14081C6.3537 7.33974 6.04223 7.55596 5.7669 7.79986C5.25753 8.24268 4.83593 8.80486 4.50037 9.4812C4.1648 10.097 3.90496 10.8668 3.69846 11.7957C3.60381 12.2385 3.54358 12.6536 3.52638 13.0515L3.75008 12.7401C3.68125 13.1553 3.63823 13.6154 3.62963 14.1101C3.73287 13.9285 3.80171 13.8074 3.85333 13.7382L3.82752 14.9335L4.31968 13.7382C4.35409 13.842 4.37991 13.9544 4.41432 14.0841C4.71547 13.3819 4.97531 12.8024 5.20763 12.3423C5.43994 11.8493 5.68257 11.4151 5.91489 11.0432C6.13859 10.6713 6.39844 10.3063 6.67377 9.96035C6.92329 9.64899 7.24337 9.27536 7.63055 8.86886C7.69939 8.77372 8.1055 8.5921 8.13131 8.58345C8.3034 8.52291 8.5185 8.45372 8.79555 8.38452C8.92462 8.34993 9.23609 8.28074 9.70931 8.15965C10.1223 8.06452 10.4596 7.96938 10.7177 7.90883C11.5196 7.68396 12.1322 7.44006 12.5711 7.1806C13.132 6.83464 13.459 6.43506 13.5364 5.96802C12.9927 6.08911 12.4506 6.16695 11.8896 6.20154C11.5885 6.21884 11.1393 6.22749 10.5267 6.22749C9.99153 6.22749 9.57853 6.23614 9.28599 6.27073C8.83685 6.32263 8.39804 6.40912 7.97472 6.5648C7.49977 6.72567 7.07817 6.91594 6.69786 7.14081Z"
        fill="currentColor"
      />
    </svg>
  </div>
</div>
`;
